File: app/src/main/java/mm/pndaza/tipitakamyanmar/repository/TocRepository.java
package mm.pndaza.tipitakamyanmar.repository;

import android.database.Cursor;
import android.database.sqlite.SQLiteOpenHelper;

import java.util.ArrayList;

import mm.pndaza.tipitakamyanmar.model.Toc;
import mm.pndaza.tipitakamyanmar.utils.SQLBuilder;

public class TocRepository extends BaseRepository {

    public TocRepository(SQLiteOpenHelper dbHelper) {
        super(dbHelper);
    }

    private static final String TOC_TABLE = "toc";
    private static final String TOC_NAME_COLUMN = "name";
    private static final String TOC_TYPE_COLUMN = "type";
    private static final String BOOK_ID_COLUMN = "book_id";
    private static final String PAGE_NUMBER_COLUMN = "page_number";

    public ArrayList<Toc> getToc(String bookId) {
        ArrayList<Toc> tocArrayList = new ArrayList<>();
        final SQLBuilder sqlBuilder = new SQLBuilder().
                select(TOC_NAME_COLUMN, TOC_TYPE_COLUMN, PAGE_NUMBER_COLUMN).
                from(TOC_TABLE).
                where(BOOK_ID_COLUMN , " = ", bookId);
        final String sql = sqlBuilder.build();
        try (Cursor cursor = getReadableDatabase().rawQuery(sql, null)) {
            if (cursor != null && cursor.moveToFirst()) {
                do {
                    String name = cursor.getString(cursor.getColumnIndexOrThrow(TOC_NAME_COLUMN));
                    int type = cursor.getInt(cursor.getColumnIndexOrThrow(TOC_TYPE_COLUMN));
                    int pageNumber = cursor.getInt(cursor.getColumnIndexOrThrow(PAGE_NUMBER_COLUMN));
                    tocArrayList.add(new Toc(String.valueOf(type), name, pageNumber));
                } while (cursor.moveToNext());
            }
        }
        return tocArrayList;
    }
}

File: app/src/main/java/mm/pndaza/tipitakamyanmar/repository/TocRepository.kt
package mm.pndaza.tipitakamyanmar.repository

import android.database.sqlite.SQLiteOpenHelper
import mm.pndaza.tipitakamyanmar.database.DatabaseSchema.TocTable
import mm.pndaza.tipitakamyanmar.model.Toc

class TocRepository(dbHelper: SQLiteOpenHelper) : BaseRepository(dbHelper) {
    fun getToc(bookId: String?): ArrayList<Toc> {
        val tocArrayList = ArrayList<Toc>()
        val sql = """
            SELECT ${TocTable.name}, ${TocTable.type}, ${TocTable.pageNumber}
             FROM ${TocTable.table}
             WHERE ${TocTable.bookId} = ?
        """.trimIndent()
        val selectionArgs = arrayOf(bookId)
        readableDatabase.rawQuery(sql, selectionArgs).use { cursor ->
            if (cursor.moveToFirst()) {
                do {
                    val name = cursor.getString(cursor.getColumnIndexOrThrow(TocTable.name))
                    val type = cursor.getInt(cursor.getColumnIndexOrThrow(TocTable.type))
                    val pageNumber = cursor.getInt(cursor.getColumnIndexOrThrow(TocTable.pageNumber))
                    tocArrayList.add(Toc(type.toString(), name, pageNumber))
                } while (cursor.moveToNext())
            }
        }
        return tocArrayList
    }

}